hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -3736,13 +3755,17 @@ File replaceOMDBWithCheckpoint(long lastAppliedIndex, File oldDB,
       // starting up.
       Files.createFile(markerFile);
       FileUtils.moveDirectory(checkpointPath, oldDB.toPath());
+      moveOmSnapshotData(oldDB.toPath(), dbSnapshotsDir.toPath());
```

Comment:
Getting all of snapshot data and installing it from leader can take a long time. We should at least have the logic to skip getting the snapshot directories from leader that already exist on the follower.

Reply:
This would work even when we have thousands of snapshots as the process should eventually converge.

Reply:
Yes, but to do so I would need to add leader tracking as well.  At that point, I will just be redoing all the work that @Xushaohong has already done more elegantly in the incremental snapshot pr here:  https://github.com/apache/ozone/pull/4294

Reply:
No, but @prashantpogde and I agreed to hold off on this because this PR is already too big.  I've added this ticket as a reminder to finish it: https://issues.apache.org/jira/browse/HDDS-8208

Reply:
Got it.